music_assistant/common/models/config_entries.py

```diff
@@ -254,7 +257,8 @@ def update(self, update: dict[str, ConfigValueType]) -> set[str]:
             changed_keys.add(key)
 
         # config entry values
-        values = update.get("values", update)
+        values = cast(dict[str, ConfigValueType], update.get("values", update))
```

Comment:
A config object.dict has values with the actual configentry values.

Reply:
So really ConfigValueType needs to include `dict[str, ConfigValueType]` as an allowed type (if they can be recursive).

Reply:
What i'm stuck on here is that update() is typed as `dict[str, ConfigValueType]`, but that doesn't seem to be the case. Because the dict passed in has a key which itself is a `dict[str, ConfigValueType]`, which is why i needed the cast to get this line passing.

So i really need a TypedDict here that has enabled, name and values maybe?

Reply:
d598c3cfb0fadc31c2d25d99bfa7ce7fe015846b seems to work. I could not find a case where it broke, but.. Is it right??

Reply:
We're actually cheating in the frontend and secretly send the enabled and name within the values ;-)
https://github.com/music-assistant/frontend/blob/main/src/views/settings/EditPlayer.vue#L115

So I think we should either fix that or create an extra type

Reply:
Let's keep it this way and fix the api to send enabled and name as separate params instead of hiding them in the config values. Either that or make enabled and name actual config entries. Now its a bit mangled which I agree is not great, your confusion here in the typing is a proof of that.